src/wdlci/utils/write_workflow.py

```diff
@@ -94,6 +97,29 @@ def write_workflow(
         struct_typedefs ([WDL.Env.Binding]): structs imported by the main workflow; these will be available to the test task
         custom_test_dir (str): Path to a directory containing test WDL tasks; this directory will be checked for test tasks first
     """
+
+    # Create an instance of the config and create a list of all the missing outputs
+    config = Config.instance()
+    all_outputs = []
+    output_file_name_pattern = re.compile(r"\b\w+\s+(\S+)\s+=")
+    missing_outputs_list = []
+
+    for workflow, workflow_config in config.file.workflows.items():
```

Comment:
It looks like you're reading every WDL file defined in the config _every single time you write a workflow for a task output_.

I think a better place for this code would be somewhere in `submit.py`, so that at minimum the code that grabs the list of all outputs is only run once, then before you even call `write_workflow()` you confirm that all the outputs expected match up - i.e. you don't enter `write_workflow` at all if some output is mismatched.

Ideally rather than checking if a task's output is present in a list of all outputs from all tasks you would specifically check whether the outputs of _that task_ in _that workflow file_ are identical to the outputs in the config file.

The other benefit to doing this in `submit.py` is that you'll already have handles for the workflow files and for the config file that you can access directly.